tools/harness-simulation/posix/launch_testbed.py

```diff
@@ -155,15 +174,21 @@ def main():
     args = parser.parse_args()
 
     # Check validation of arguments
-    if not 0 <= args.ot11_num <= MAX_OT11_NUM:
-        raise ValueError(f'The number of FTDs should be between 0 and {MAX_OT11_NUM}')
+    if args.ot_num < 0:
+        raise ValueError(f'The number of FTDs should be non-negative')
+
+    if args.otbr_num < 0:
+        raise ValueError(f'The number of OTBRs should be non-negative')
 
     if not 0 <= args.sniffer_num <= MAX_SNIFFER_NUM:
-        raise ValueError(f'The number of FTDs should be between 0 and {MAX_SNIFFER_NUM}')
+        raise ValueError(f'The number of sniffers should be between 0 and {MAX_SNIFFER_NUM}')
 
-    if args.ot11_num == args.sniffer_num == 0:
+    if args.ot_num == args.otbr_num == args.sniffer_num == 0:
         raise ValueError('At least one device is required')
 
+    if args.ot_num + args.otbr_num > MAX_NODES_NUM:
+        raise ValueError(f'The number of all devices should be no more than {MAX_NODES_NUM}')
```

Comment:
Use assert?

Reply:
These numbers are specified by the user. Therefore I think `raise` is more appropriate here.

The *Goole Python Style Guide* says
> Do not use assert statements for validating argument values of a public API. assert is used to ensure internal correctness, not to enforce correct usage nor to indicate that some unexpected event occurred. If an exception is desired in the latter cases, use a raise statement.